internal/cache/file/cache_handle_test.go

```diff
@@ -820,3 +820,31 @@ func (cht *cacheHandleTest) Test_MultipleReads_CacheHitShouldBeFalseThenTrue() {
 	assert.True(cht.T(), cacheHit)
 	assert.Nil(cht.T(), err)
 }
+
+func (cht *cacheHandleTest) Test_Read_Sequential_Parallel_Download_True() {
+	dst := make([]byte, ReadContentSize)
+	offset := int64(cht.object.Size - ReadContentSize)
+	cht.cacheHandle.isSequential = true
+	cht.cacheHandle.cacheFileForRangeRead = true
```

Comment:
let's keep to cacheFileForRangeRead = false in this case. This will ensure that downloads are triggered in case of Sequential reads and Parallel downloads.

Reply:
Yes, it should be false , nice catch !!!